js/main.js

```diff
@@ -0,0 +1,53 @@
+const NAMES_ARRAY = [
+  'Андрей','Мария',
+  'Станислав',
+  'Олег'
+];
+const COMMENTS_ARRAY = [
+  'Всё отлично!',
+  'В целом всё неплохо. Но не всё. Когда вы делаете фотографию, хорошо бы убирать палец из кадра.',
+  'Моя бабушка случайно чихнула с фотоаппаратом в руках и у неё получилась фотография лучше.'
+];
+const DESCRIPTION = [
+  'Это мяч',
+  'Коты наше все',
+  'Анапа 2008',
+  '#Я в коридоре на Балли'
+];
+const OBJ = [];
+const getRandomPositiveInteger = (a, b) =>{
+  const lower = Math.ceil(Math.min(a, b));
+  const upper = Math.floor(Math.max(a, b));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+const createIdGenerator = () => {
+  let lastGeneratedId = 0;
+  return function (){
+    lastGeneratedId += 1;
+    return lastGeneratedId;
+  };
+};
+const randomIdNumber = createIdGenerator();
+const randomUrlNumber = createIdGenerator();
+const createObj = () => {
+  const randomDescriptionIndex = getRandomPositiveInteger(0, DESCRIPTION.length - 1);
+  const randomCounterLikes = getRandomPositiveInteger(15,200);
+  const randomCommentsIndex = getRandomPositiveInteger(0,COMMENTS_ARRAY.length - 1);
+  const randomNameIndex = getRandomPositiveInteger(0,NAMES_ARRAY.length - 1);
+  return {
+    description: DESCRIPTION[randomDescriptionIndex],
+    likes: randomCounterLikes,
+    comments: COMMENTS_ARRAY[randomCommentsIndex],
```

Comment:
comments -- должен быть массивом объектов. Нужно, к примеру, вынести их генерацию в отдельную функцию выше, а тут её вызывать
```suggestion
    comments: createComments(),
```

Reply:
Сама функция `createComments` идейно последней функции должна собирать массив определенной в константах выше длины `COMMENTS_COUNT` с объектами каждого комментария, типо вот этого:
```ts
const createComments = () => {
  const comments = []
   for(let i = 0; i < COMMENTS_COUNT; i++){
    comments [i] = {
         id: /*тут уникальный id*/,
        avatar: `img/avatar-${getRandomPositiveInteger(0, 6)}.svg`,
        message: getRandomArrayElement(MESSAGES)
       name: getRandomArrayElement(NAMES),
    };
  }
  return comments
}
```

Конечно, можно это ещё разделить на несколько функций или использовать Array.from, например так:
```ts
const createComments = () => Array.from({length: COMMENTS_COUNT}, () => ({
  id: /*тут уникальный id*/,
  avatar: `img/avatar-${getRandomPositiveInteger(0, 6)}.svg`,
  message: getRandomArrayElement(MESSAGES)
  name: getRandomArrayElement(NAMES),
}))
```

Или так:
```ts
const createComment = () => ({
  id: /*тут уникальный id*/,
  avatar: `img/avatar-${getRandomPositiveInteger(0, 6)}.svg`,
  message: getRandomArrayElement(MESSAGES)
  name: getRandomArrayElement(NAMES),
})

const createComments = () => Array.from({length: COMMENTS_COUNT}, createComment)
```

Главное, чтобы было понимание, что у нас будет функция `createComments` которая возвращает массив вот такого вида: 
![image](https://user-images.githubusercontent.com/11852682/222149177-03044cc7-e556-4005-aa8d-6c3c8985e812.png)